storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3552,6 +3552,9 @@ void fsp_system_tablespace_truncate()
 {
   uint32_t last_used_extent= 0;
   fil_space_t *space= fil_system.sys_space;
+
+  if (!space) return;
```

Comment:
Why do we need this condition? The only reason that I can think of would be that the server failed to start up for whatever reason. What would be the preceding error log messages in that case? What is the value of `srv_was_started`?

Wouldn’t it be safer to call this function in `innobase_end()` right before `innodb_shutdown()`?